### ISSUE TITLE
Properly handle .git folders

### ISSUE THREAD

**Issue by callegar** [hbons/SparkleShare#335]
Currently sparkleshare inherits all the features/limitations/quirks of its backend,

Consequently, it does not sync empty dirs (issue 326 - https://github.com/hbons/SparkleShare/issues/326) and it does not sync directories named .git. Furthermore, I expect it to get confused if you have a file named .gitignore, or .gitattributes or to simply ingnore it (haven't tried).

Conversely, I think that sparkleshare should hide the backend specific characteristics.

Particularly, when the backend is git, it should

1) on the way from the client to the server, put a placeholder in empy dirs and remove it on the way from the server to the client
2) on the way from the client to the server rename all .git<something> files to .git<something><random_key> and remove the <random_key> on the way back.

Most likely the above issues exist also for the mercurial backend, with sparkleshare being unable to deal with .hg directories (or whatever mercurial has). 

In conclusion, any backend should probably be able to specify push and pull rules to hide the peculiarities it brings in:

E.g. (for git)

On empty directory:
- PUSH create file placeholder_<random_key>
- PULL remove file placeholder_<random_key>

On file/dir with name matching pattern .git*
- PUSH rename file/dir to name_<random_key>
- PULL rename file/dir named name_<random_key> into name

**Comment by alazare619** [comment]
Is this still a issue?
